modules/programs/mangohud.nix

```
@@ -0,0 +1,103 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.mangohud;
+
+  renderOption = option:
+    rec {
+      int = toString option;
+      float = int;
+      path = int;
+      bool = "false";
+      string = option;
+      list = concatStringsSep "," option;
+    }.${builtins.typeOf option};
+
+  renderLine = k: v: (if isBool v && v then k else "${k}=${renderOption v}");
+  renderSettings = attrs:
+    strings.concatStringsSep "\n" (attrsets.mapAttrsToList renderLine attrs)
+    + "\n";
+
+in {
+  options = {
+    programs.mangohud = {
+      enable = mkEnableOption "Mangohud";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.mangohud;
+        defaultText = literalExample "pkgs.mangohud";
+        description = "The Mangohud package to install.";
+      };
+
+      enableSessionWide = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Sets environment variables so that 
+          MangoHud is started on any application that supports it.
+        '';
+      };
+
+      settings = mkOption {
+        type = with types;
+          attrsOf (oneOf [ bool int float str path (listOf str) ]);
+        default = { };
+        example = literalExample ''
+          {
+            output_folder = ~/Documents/mangohud/;
+            full = true;
+          }
+        '';
+        description = ''
+          Configuration written to
+          <filename>~/.config/MangoHud/MangoHud.conf</filename>. See
+          <link xlink:href="https://github.com/flightlessmango/MangoHud/blob/master/bin/MangoHud.conf"/>
+          for the default configuration.
+        '';
+      };
+
+      settingsPerApplication = mkOption {
+        type = with types; attrsOf attrs;
+        default = { };
+        example = literalExample ''
+          {
+            mpv = {
+              no_display = true;
+            }
+          }
+        '';
+        description = ''
+          Sets MangoHud settings per application.
+          Configuration written to
+          <filename>~/.config/MangoHud/{application_name}.conf</filename>. See
+          <link xlink:href="https://github.com/flightlessmango/MangoHud/blob/master/bin/MangoHud.conf"/>
+          for the default configuration.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.package ];
+
+      home.sessionVariables = mkIf cfg.enableSessionWide {
+        MANGOHUD = 1;
+        MANGOHUD_DLSYM = 1;
+      };
+
+      xdg.configFile."MangoHud/MangoHud.conf" =
+        mkIf (cfg.settings != { }) { text = (renderSettings cfg.settings); };
```

Unnecessary parentheses:
```suggestion
        mkIf (cfg.settings != { }) { text = renderSettings cfg.settings; };
```